File: Python/Greedy+ Sorting/Partition Array Such That Maximum Difference Is K.py
class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()
        cnt=0
        mini=nums[0]
        for i in range(len(nums)):
            if nums[i]-mini>k:
                cnt+=1
                mini=nums[i]
        return cnt+1
        

File: CPP/Sorting/Partition Array Such That Maximum Difference Is K.cpp
class Solution {
public:
    int partitionArray(vector<int>& nums, int k) {
        sort(nums.begin(),nums.end());
        int cnt=0;
        int mini=nums[0];
        for(int i=0;i<nums.size();i++){
            if(nums[i]-mini>k){
                cnt++;
                mini=nums[i];
            }
        }
        // +1 bcz counting the first group
        return cnt+1;
    }
};
